pinot-server/src/main/java/org/apache/pinot/server/api/resources/HealthCheckResource.java

```diff
@@ -54,8 +56,13 @@ public class HealthCheckResource {
       @ApiResponse(code = 200, message = "Server is healthy"),
       @ApiResponse(code = 503, message = "Server is not healthy")
   })
-  public String checkHealth() {
-    return getReadinessStatus(_instanceId);
+  public String checkHealth(
+      @ApiParam(value = "health check type: liveness or readiness") @QueryParam("checkType") String checkType) {
+    if (checkType == null || checkType.equalsIgnoreCase("readiness")) {
```

Comment:
Suggest doing liveness only when it is explicitly configured to prevent unexpected behavior
```suggestion
    if ("liveness".equalsIgnoreCase(checkType)) {
      return "OK";
    } else {
      return getReadinessStatus(_instanceId);
    }
```